### PR TITLE
fish-style shortened cwd as short_cwd

### PR DESCRIPTION
**Pull request by adqm**
Not sure whether this should just stay at home in my `.xonshrc` or whether it's something worth including...

I do a lot of work deep in directory trees, and so I found fish's shortening of the CWD in the prompt to be useful.  This changeset implements a shortened CWD as a new entry in the formatter dict, `short_cwd`.

If it's not useful (or too adqm-specific) to have this in the main codebase, I'm happy to close this and just leave it to live on my own machines, but I thought it might be useful.

### CODE REVIEW

File: xonsh/environ.py [hunk 1]
```diff
@@ -539,6 +539,13 @@ def _replace_home(x):
 
 _replace_home_cwd = lambda: _replace_home(builtins.__xonsh_env__['PWD'])
 
+def _collapsed_pwd():
+    pwd = _replace_home_cwd().split('/')
```
Comment (scopatz) [inline xonsh/environ.py]: I think that `'/'` should be replaced with `os.sep` everywhere in this function.

**Comment by scopatz** [comment]
Thanks @adqm - I think that if you find this useful, other people are bound to as well.  Also, could you add this as an entry to the tutorial docs so that folks know about it.

**Comment by adqm** [comment]
Okay; tested this on GNU/Linux and on Windows (both with and without `$FORCE_POSIX_PATHS` set), so I think it is ready for another look.

**Comment by scopatz** [comment]
Looks great to me, thanks @adqm!

**Pull request merged**